packages/flutter_tools/lib/src/build_runner/web_fs.dart

```diff
@@ -319,7 +315,11 @@ class WebFs {
       } else if (request.url.path.contains('assets')) {
         final String assetPath = request.url.path.replaceFirst('assets/', '');
         final File file = fs.file(fs.path.join(getAssetBuildDirectory(), assetPath));
-        return Response.ok(file.readAsBytesSync());
+        try {
+          return Response.ok(file.readAsBytesSync());
+        } on FileSystemException {
+          return Response.notFound('');
```

Comment:
Should we do `if (file.existsSync())` instead of a catch-all? We probably don't want to lose exception information for things we do not expect to happen.

Reply:
Right! we can check for files before we read them ... I forgot that. Fixed